toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go

```diff
@@ -676,3 +660,42 @@ func (mp *ManualPartitionWidget) onNextButton() {
 		mp.nextPage()
 	}
 }
+
+// enumInputBox returns an input box that only allows values from elements to appear.
+// As well as resetter to reset the selection
+func (mp *ManualPartitionWidget) enumInputBox(elements []string) (field *tview.InputField, resetter func()) {
+	field = tview.NewInputField()
+	index := 0
+	// Initialize text with the first element
+	field.SetText(elements[index])
+
+	field.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
+		key := event.Key()
+		// Override movement left/right. Leave navigation keys intact. Consume all other input.
+		switch key {
+		case tcell.KeyLeft:
+			if index == 0 {
+				index = len(elements)
+			}
+			index--
```

Comment:
Suggestion: this and `tcell.KeyRight` look like good places for a modulo operation `index = (elementsCount + index + [change]) % elementsCount`.